src/main/kotlin/com/natpryce/konfig/magic.kt

```diff
@@ -14,10 +14,15 @@ open class PropertyKeys : Iterable<Key<*>> {
 
 open class PropertyGroup(private val outer: PropertyGroup? = null) : PropertyKeys() {
     private fun outer() = outer ?: javaClass.enclosingClass?.kotlin?.objectInstance as? PropertyGroup
-    private fun name() : String = namePrefix() + groupName()
-    private fun namePrefix() = outer()?.name()?.let { it + "." } ?: ""
-    private fun groupName() = javaClass.kotlin.simpleName?.substringBefore("$") ?:
-            throw IllegalArgumentException("cannot determine name of property group")
+    private fun name(): String = namePrefix() + groupName()
+    private fun namePrefix() = outer()?.name()?.let { "$it." } ?: ""
+
+    // anonymous objects don't get names in their KClass, so we have to resort to the JVM's name
+    private fun groupName() = javaClass.kotlin.simpleName?.substringBefore("$")
+        ?: javaClass.name.split("$").let { chunks ->
```

Comment:
turns out `simpleName` (and the other names in `KClass`) were documented to be null for anonymous objects, but happened to be non-null before. Now they are null, so this will fall back to attempting to parse `Class.getName()` before giving up.